### ISSUE TITLE
Possible to test another validation set that have   labels diffierent from train set and test set?

### ISSUE THREAD

**Issue by andyhx** [BVLC/caffe#5765]
hi,I want to test another dataset(assuming A) when training the network, I want to callucate the loss and accurate rate of the model,this dateset is different labels from trainset and testset ,it is not serapated  from the dateset divided into  two trainset and testset
so  is it right  to just add this dataset A into the trainval.prototxt by adding the  test stages  in the phase: TEST, or i have to modify the  caffe testnet code  to test A set and rebuild the caffe to do this ,so is there a  more simple way or an example to follow ,thank  you in avdance!

**Comment by andyhx** [comment]
anyone knows , I have been google for a long time ,but I cant find any cues

**Comment by andyhx** [comment]
any guide?

**Comment by zhaifly** [comment]
did you use the new test dataset(A) to training the network?
or you want to use a trained network to test your new dataset?
what the difference between the two lables?

**Comment by andyhx** [comment]
hi,@zhaifly ,thanks for ur reply, 
1.I dont uset the new test dataset(A)  to train the network
2.Yes I want to test the A dataset's FRR and ERR result , when training  not after the train finished ;
3.the A lables is index from   0-100,train and test is 0-2000,their index  is referece to anthoer  persons ,It is a new dataset ,not same as train and test dataset!

**Comment by zhaifly** [comment]
@andyhx  is there has any relation between labels(0~100) and labels(0~2000), if has, i think you can map the labels(0~100) to labels(0~2000) manually. if not, i think you need to modify your network and use your new data to train it and then test it.

i am new to caffe and DL, may it help you!

**Comment by andyhx** [comment]
@zhaifly  thank u,there is no connections between them,I think training a new  data is not a good idea,So
 anyone has come across the same problem,help me ,thank u!

**Comment by Noiredd** [comment]
You will need a different output layer - the training output is has 2000 probabilities, while for testing you would like to have just 100. This is a question about what exactly are you trying to do, and what is your expected outcome by testing on a whole another data. What does it mean that the labels in dataset A are all different?

Please post this question to the [caffe-users list][1] - all usage, installation, or modeling questions, or other requests for help should go there instead of Issues. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe. Please read the [guidelines for contributing][2] before submitting an issue or a pull request.

 [1]: https://groups.google.com/forum/#!forum/caffe-users
 [2]: https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md
